File: Python3/575. Distribute Candies.py
class Solution:
    def distributeCandies(self, candies):
        """
        :type candies: List[int]
        :rtype: int
        """
        # return int(len(candies)/2 if len(set(candies)) > len(candies)/2 else len(set(candies)))
        return int(min(len(candies)/2, len(set(candies))))

File: Golang/575. Distribute Candies.go
// func distributeCandies(candies []int) int {
//     hashtable := make(map[int]int, len(candies))
//     for _, candy := range candies {
//         hashtable[candy]++
//     }
//     if len(hashtable) > len(candies)/2 {
//         return len(candies)/2
//     } else {
//         return len(hashtable)
//     }
// }

func distributeCandies(candies []int) int {
    n := len(candies)
    hashtable := make(map[int]bool, n)
    
    for _, candy := range candies {
        hashtable[candy] = true
    }
    
    n2 := len(hashtable)
    
    if n2 > n/2 {
        return n/2
    } else {
        return n2
    }
}
